src/GafferScene/ShaderTweaks.cpp

```diff
@@ -316,8 +323,66 @@ bool ShaderTweaks::applyTweaks( IECoreScene::ShaderNetwork *shaderNetwork, Tweak
 				{
 					throw IECore::Exception( fmt::format( "Cannot apply tweak to \"{}\" : Mode must be \"Replace\" when inserting a connection", name ) );
 				}
+
 				const auto inputParameter = ShaderNetworkAlgo::addShaders( shaderNetwork, inputNetwork );
 				shaderNetwork->addConnection( { inputParameter, parameter } );
+
+				// TODO - it would be substantially more efficient to search for and process tweak sources
```

Comment:
Yeah, technically I don't think we're even privy to the handle uniquefication mechanism either, so we can't confidently predict the handles. Feels like we want an optional output `unordered_map<oldHandle, newHandle>` from `addShaders()`?

Reply:
Left this note in, with mention of your addShaders optional out proposal, but no longer a TODO. At least now that we have the blindData for whether proxies exist, there's no extra cost here in existing scenes